align_system/algorithms/outlines_hybrid_regression_adm.py

```diff
@@ -0,0 +1,215 @@
+    sentences = load_itm_sentences(data_dict)
+    sentences = ["[CLS] " + s for s in sentences]
+    tokenizer = AutoTokenizer.from_pretrained(model_name)
+    ids, amasks = get_ids_mask(sentences, tokenizer, max_length=512)
+    inputs, masks = torch.tensor(ids), torch.tensor(amasks)
+    data = TensorDataset(inputs, masks)
+    return data
+
+
+def get_kdma_predictions(model_name: str, data_dict: Dict[str, List], model_path: str) -> np.ndarray:
+    dataset = load_model_data(
+        model_name=model_name,
+        data_dict=data_dict
+    )
+    dataloader = DataLoader(dataset, batch_size=1)
+    regression_model = BertRegressionModel(model_name=model_name, load_path=model_path)
+    predictions = regression_model.evaluate(dataloader)
+
+    return predictions
+
+
+class BertRegressionModel:
+    def __init__(self, model_name: str, load_path: str,
+                 device="cuda" if torch.cuda.is_available() else "cpu"):
+        self.model_name = model_name
+        self.load_path = load_path
+        self.device = device
+        config = AutoConfig.from_pretrained(self.model_name, num_labels=1)
+        self.model = AutoModelForSequenceClassification.from_pretrained(self.model_name, config=config)
+
+    def evaluate(self, dataloader):
+        self.model.load_state_dict(torch.load(self.load_path))
+        self.model.to(self.device)
+        self.model.eval()
+        predictions = []
+        for batch in dataloader:
+            # Copy data to GPU if needed
+            if self.device == "cuda":
+                batch = tuple(t.cuda() for t in batch)
+            else:
+                batch = tuple(t.cpu() for t in batch)
+
+            # Unpack the inputs from our dataloader
+            b_input_ids, b_input_mask = batch
+
+            # Forward pass
+            with torch.no_grad():
+                logits = self.model(b_input_ids, attention_mask=b_input_mask)[0]
+            output = logits.squeeze().detach().cpu().numpy()
+            predictions.append(output)
+        predictions = np.clip(output, 0, 1)
+
+        return predictions
+
+
+class HybridRegressionADM(OutlinesTransformersADM):
+    def __init__(self,
+                 model_name,
+                 device='auto',
+                 baseline=False,
+                 sampler=MultinomialSampler(),
+                 **kwargs):
+        self.baseline = baseline
+        self.model = outlines.models.transformers(
+            model_name,
+            device=device,
+            model_kwargs=kwargs.get('model_kwargs', {}),
+            tokenizer_kwargs=kwargs.get('tokenizer_kwargs', {}))
+        # NOTE: In cases where we want multiple samples, we're passing
+        # in a list of prompts (this allows us to shuffle answers in
+        # each prompt), rather than setting the number of samples in
+        # the sampler itself (which defaults to 1); setting the number
+        # of samples in the sampler may result in unexpected behavior
+        self.sampler = sampler
```

Comment:
I think we should just let this be inherited if you don't need to do anything in particular for your approach here